src/app/portlets/dot-edit-page/content/dot-edit-content.component.ts

```diff
@@ -162,6 +168,33 @@ export class DotEditContentComponent implements OnInit {
             });
     }
 
+    /**
+     * Hanlde changes in the configuration of "View As" toolbar
+     *
+     * @param {DotEditPageViewAs} viewAsConfig
+     * @memberof DotEditContentComponent
+     */
+    changeViewAsHandler(viewAsConfig: DotEditPageViewAs): void {
+        this.route.queryParams.subscribe(params => {
```

Comment:
When you need something from the first Observable to call the second one you should use ```concatMap```